src/sql/workbench/contrib/query/browser/queryInputFactory.ts

```diff
@@ -82,16 +87,20 @@ export class QueryEditorLanguageAssociation implements ILanguageAssociation {
 			return undefined;
 		}
 
-		const profile = getCurrentGlobalConnection(this.objectExplorerService, this.connectionManagementService, this.editorService);
-		if (profile) {
-			const options: IConnectionCompletionOptions = {
-				params: { connectionType: ConnectionType.editor, runQueryOnCompletion: undefined, input: queryEditorInput },
-				saveTheConnection: false,
-				showDashboard: false,
-				showConnectionDialogOnError: true,
-				showFirewallRuleOnError: true
-			};
-			this.connectionManagementService.connect(profile, queryEditorInput.uri, options).catch(err => onUnexpectedError(err));
+		const existingProfile = this.connectionManagementService.getConnectionProfile(queryEditorInput.uri);
```

Comment:
Would be good to just put these both in a helper function since they seem to be doing the same thing